File: src/main/java/io/dustin/finstage/financial/domain/CashFlowStatement.java
package io.dustin.finstage.financial.domain;

import lombok.Builder;
import lombok.Getter;

import java.time.LocalDate;

@Getter
@Builder
public class CashFlowStatement {
    private final String symbol;
    private final LocalDate date;
    private final long operatingCashFlow;
    private final long investingCashFlow;
    private final long financingCashFlow;
    private final long freeCashFlow;

    private CashFlowStatement(
            String symbol,
            LocalDate date,
            long operatingCashFlow,
            long investingCashFlow,
            long financingCashFlow,
            long freeCashFlow
    ) {
        this.symbol = symbol;
        this.date = date;
        this.operatingCashFlow = operatingCashFlow;
        this.investingCashFlow = investingCashFlow;
        this.financingCashFlow = financingCashFlow;
        this.freeCashFlow = freeCashFlow;
    }

    public static CashFlowStatement of(
            String symbol,
            LocalDate date,
            long operatingCashFlow,
            long investingCashFlow,
            long financingCashFlow,
            long freeCashFlow
    ) {
        return new CashFlowStatement(symbol, date, operatingCashFlow, investingCashFlow, financingCashFlow, freeCashFlow);
    }
}

File: src/main/kotlin/io/dustin/finstage/financial/domain/CashFlowStatement.kt
package io.dustin.finstage.financial.domain

import java.time.LocalDate

data class CashFlowStatement(
        val symbol: String,
        val date: LocalDate,
        val operatingCashFlow: Long,
        val investingCashFlow: Long,
        val financingCashFlow: Long,
        val freeCashFlow: Long
) {
    companion object {
        fun of(
                symbol: String,
                date: LocalDate,
                operatingCashFlow: Long,
                investingCashFlow: Long,
                financingCashFlow: Long,
                freeCashFlow: Long
        ): CashFlowStatement {
            return CashFlowStatement(
                    symbol,
                    date,
                    operatingCashFlow,
                    investingCashFlow,
                    financingCashFlow,
                    freeCashFlow
            )
        }
    }
}
